examples/kdv_1d/kdv_1d_non_dim.jl

```diff
@@ -0,0 +1,60 @@
+using OrdinaryDiffEqTsit5
+using DispersiveShallowWater
+using SummationByPartsOperators: upwind_operators, periodic_derivative_operator
+
+###############################################################################
+# Semidiscretization of the KdV equation 
+
+function initial_condition_non_dimensional(x, t, equations::KdVEquation1D, mesh)
+    c = 1 / 3
+    x_t = mod(x - c * t - xmin(mesh), xmax(mesh) - xmin(mesh)) + xmin(mesh)
+    u = 3 * c * sech(sqrt(9 * c) / 6 * x_t)^2
+    return u
+end
+
+function initial_condition_non_dimensional_converted(x, t, equations::KdVEquation1D, mesh)
+    u = initial_condition_non_dimensional(x, t, equations, mesh)
+    eta = nondim2prim(u, equations)
+    return SVector(eta)
```

Comment:
I would prefer calling `nondim2prim` on an `SVector` `u` instead of a scalar because in general these conversion functions usually assume an `SVector` input and also return an `SVector` output. I would like to keep this pattern also for `nondim2prim` and `prim2nondim`. I think it's fine how you define the functions below allowing both vectors and scalars, but I would still only call them on `SVector`s for consistency. See my suggestion above.
```suggestion
    return nondim2prim(u, equations)
```